docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+Changes not saved; Run with --write to commit to disk
+```
+
+### Manually update deprecated API references
+
+SilverStripe 4 introduces thousands of small and big API changes. To update deprecated API references manually, you have to go through each one of your project files.
+
+[Read the SilverStripe 4 change logs](/changelogs/4.0.0/) for a comprehensive list of what has changed.
+
+### Finalising the deprecated API update
+At this stage, your site should be using only SilverStripe 4 API logic.
+
+You still have some minor clean up tasks and configuration tweaks to apply, but you're almost done.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 6 - Update your entry point {#step6}
+The location of SilverStripe's _entry file_ has changed. Your project and server environment will need
+to adjust the path to this file from `framework/main.php` to `index.php`.
+
+### Update your entry point
+You can get a copy of the SilverStripe 4 `index.php` file at:
+* `vendor/silverstripe/recipe-core/public/index.php` if you are upgrading to SilverStripe 4.1 or above
+* `vendor/silverstripe/recipe-core/index.php` if you are upgrading to SilverStripe 4.0
+
+If you've modified your SilverStripe 3 `index.php`, you'll need to reconcile those changes with the index.php file you got from `recipe-core`. Otherwise, just use the generic `index.php` file `recipe-core` provides.
+
+Copy your new `index.php` to your project's web root. Unlike SilverStripe 3, `index.php` must be present in your web root.
+
+### Update your server configuration
+If you're using a `.htaccess` file or `web.config` file to handle your server configuration, you can get the generic SilverStripe 4 version of those file from:
+* `vendor/silverstripe/recipe-core/public` if you are upgrading to SilverStripe 4.1 or above
+* `vendor/silverstripe/recipe-core/` if you are upgrading to SilverStripe 4.0
+
+Just like `index.php`, if you've modified your server configuration file from the one that shipped with SilverStripe 3, you'll need to reconcile your changes into the version retrieve from `recipe-core`.
+
+[Refer to the installation instruction for your platform](/getting_started/installation/) if your server configuration is not managed via a `.htaccess` or `web.config` file.
+
+### Finalising the entry point upgrade
+
+At this stage, you could in theory run your project in SilverStripe 4.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 7 - Update project structure (optional) {#step7}
+SilverStripe 4.0 introduces a new recommended project structure. Adopting the recommended project structure is optional, but will become mandatory in SilverStripe 5.
```

Comment:
Was this 4.0 or 4.1?

Reply:
Not sure if the official recommendation came with 4.0 or 4.1. In practice, it will work in either case.

I changed it to `SilverStripe 4 introduces ... `